pkg/controller/jobframework/reconciler.go

```diff
@@ -537,7 +537,7 @@ func (r *JobReconciler) extractPriority(ctx context.Context, podSets []kueue.Pod
 
 func extractPriorityFromPodSets(podSets []kueue.PodSet) string {
 	for _, podSet := range podSets {
-		if len(podSet.Template.Spec.PriorityClassName) > 0 {
+		if podSet.Template != nil && len(podSet.Template.Spec.PriorityClassName) > 0 {
```

Comment:
If there is TemplateRef, the priority should come from it.

Reply:
we are still picking up a copy of the template from under the sets. Or do we need to take it out of the reconciler since we don't have a pod template created at this moment